docs/source/daml/intro/8_Dependencies.rst

```diff
@@ -0,0 +1,86 @@
+
+We can see this in action. When a DAR file gets deployed to a ledger, not all meta information is preserved. 
+
+#. Note down your main package hash from running ``inspect-dar`` above
+#. Start the project using ``daml start``
+#. Open a second terminal and run ``daml ledger fetch-dar --host localhost --port 6865 --main-package-id "887056cbb313b94ab9a6caf34f7fe4fbfe19cb0c861e50d1594c665567ab7625" -o assets_ledger.dar``, making sure to replace the hash with the appropriate one.
+#. Run ``daml damlc inspect-dar assets_ledger.dar``
+
+You'll notice two things. Firstly, a lot of the dependencies have lost their names, they are now only identifiable by hash. We could of course also create a second project ``7Composing-1.0.0`` with completely different contents so even when name and version are available, package hash is the only safe identifier.
+
+That's why over the Ledger API, all types, like templates and records are identified by the triple ``(entity name, module name, package hash)``. Your client application should know the package hashes it wants to interact with. To aid that, ``inspect-dar`` also provides a machine-readable format for the information it emits: ``daml damlc inspect-dar --json assets_ledger.dar``. The ``main_package_id`` field in the resulting JSON payload is the package hash of our project.
+
+Secondly, you'll notice that all the ``*.daml``, ``*.hi`` and ``*.hie`` files are gone. This leads us to data dependencies.
+
+Dependencies and Data Dependencies
+----------------------------------
+
+Dependencies under the ``daml.yaml`` ``dependencies`` group rely on the ``*.hi``, ``*.hie``, and ``*.daml`` files. The information in these files is crucial for dependencies like the Standard Library, which provide functions, types and typeclasses.
+
+However, as you can see above, this information isn't preserved. Furthermore, preserving this information may not even be desireable. Imagine we had built ``7Composing`` with SDK 1.100.0, and are building ``8Dependencies`` with SDK 1.101.0. An inbuilt typeclass like ``Eq`` may have changed in between so we now have two different ``==`` functions. The one from the 1.100.0 Standard Library and the one from 1.101.0. This gets messy fast, which is why the SDK does not support ``dependencies`` across SDK versions. For dependencies on contract models that were fetched from a ledger, or come from an older SDK version, there is a simpler kind of dependency called ``data-dependencies``. The syntax for ``data-dependencies`` is the same, but they only rely on the "binary" ``*.dalf`` files. The name tries to confer that the main purpose of such dependencies is to handle data: Records, Choices, Templates. The stuff one needs to use contract composability across projects.
```

Comment:
The part about `Eq` is confusing. You are explaining that it can have changed and now you have two different functions. But first, that doesn’t actually work in `dependencies`, they don’t work across SDK versions at all and second if they actually changed than you also get two different versions in `data-dependencies`.

Reply:
Made it a bit vaguer.